### PR TITLE
fix:update HTTTP error codes(200 with 201 & 400 with 409)

### LINKED ISSUE #619: Fix: refactor HTTP error codes POST /register for Users service
## Description
As a GSoC student,
I need the correct HTTP error response code to be returned accordingly by Mentorship System API in relation to POST /register request,
so that I can represent the error message accordingly to both Mentorship System and Bridge In Tech users.

## Mocks
NA

### Update [Required]
- [ ] 201: Created need to be returned instead of 200: OK if the process is successful 
- [ ] 409: Conflict need to be returned instead of 400: Bad Request if the username or email already exist

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
1 hour

### PR DESCRIPTION
**Pull request by nandini45**
### Description
Fixes #619

### Type of Change:
Code
** Code/Quality Assurance Only**
New feature (non-breaking change which adds functionality pre-approved by mentors)

### Checklist:
 My PR follows the style guidelines of this project
 I have performed a self-review of my own code or materials
 Update Postman API at /docs folder

**Code/Quality Assurance Only**
 My changes generate no new warnings
 New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by nandini45** [comment]
maya do i have to change the code in test/users too>?

**Comment by mtreacy002** [comment]
Yes, please, @nandini45 . Otherwise the test will fail. if it's check for the correct response code.

**Review by mtreacy002 — CHANGES_REQUESTED** [review]
@nandini45 , can you also please change the relevant response code inside the tests?

**Review by satya7289 — COMMENTED** [review]
Add test cases to validate it is responding correct error code.

And change only error codes for POST /register for Users service in this PR(as mentioned in issue).

1 hints:
Only `app/api/dao/user.py`, `app/api/resources/user.py`, `tests/users/test_dao.py` needs to change in this PR.

**Comment by nandini45** [comment]
Yeah i did that only.. if i did any mistake do point it out..thanks @satya7289

File: app/api/resources/admin.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 @admin_ns.route("admin/new")
 @admin_ns.response(HTTPStatus.FORBIDDEN, "%s" % messages.USER_IS_NOW_AN_ADMIN)
-@admin_ns.response(HTTPStatus.BAD_REQUEST, "%s" % messages.USER_IS_ALREADY_AN_ADMIN)
+@admin_ns.response(HTTPStatus.CONFLICT, "%s" % messages.USER_IS_ALREADY_AN_ADMIN)
```
Comment (satya7289) [inline app/api/resources/admin.py:17]: Undo this change.

File: tests/users/test_api_update_user.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ def test_update_username_already_taken(self):
             content_type="application/json",
         )
 
-        self.assertEqual(400, actual_response.status_code)
+        self.assertEqual(409, actual_response.status_code)
```
Comment (satya7289) [inline tests/users/test_api_update_user.py:49]: Undo this change.

File: app/api/dao/user.py [hunk 3]
```diff
@@ -219,7 +219,7 @@ def update_user_profile(user_id: int, data: Dict[str, str]):
 
             # username should be unique
             if user_with_same_username:
-                return messages.USER_USES_A_USERNAME_THAT_ALREADY_EXISTS, HTTPStatus.BAD_REQUEST
+                return messages.USER_USES_A_USERNAME_THAT_ALREADY_EXISTS, HTTPStatus.CONFLICT
```
Comment (satya7289) [inline app/api/dao/user.py:222]: undo this change.

File: app/api/dao/user.py [hunk 4]
```diff
@@ -94,7 +94,7 @@ def delete_user(user_id: int):
 
         if user:
             user.delete_from_db()
-            return messages.USER_SUCCESSFULLY_DELETED, HTTPStatus.OK
+            return messages.USER_SUCCESSFULLY_DELETED, HTTPStatus.CREATED
```
Comment (satya7289) [inline app/api/dao/user.py:97]: Undo this change.

**Review by satya7289 — COMMENTED** [review]
Update swagger and returned status with `created` and `conflict` error status for `/register`  (`app/api/resources/user.py` `UserRegister` Class)
And write testcase to verify this error code(`tests/users/test_dao.py`).

File: app/api/dao/user.py (hunk 3 above)
Comment (nandini45) [reply app/api/dao/user.py:222]: but it is according to maya said,
409 has to be written when user email already exists

**Comment by nandini45** [comment]
image

 In resource/users how can i changes the code for all four messgas. if i try n make this change it shows error 

image

and in tests/users/test_dao i added the code 
self.assertEqual(409, actual_response.status_code)

**Comment by mtreacy002** [comment]
cc @anitab-org/bridgeintech-maintainers or @anitab-org/coding-team , can you please help resolve this? Thanks

**Comment by isabelcosta** [comment]
@nandini45 do you still want to work on this issue? If not we can reassign, if you need help please let us know as this is an important issue for BIT.
cc @anitab-org/bridgeintech-maintainers

**Comment by nandini45** [comment]
i opened a new pull request, another member told me to open a new pull request to resolve this issue, as this had unnecessary commits that's why
